### PR TITLE
AAP-19988

### PR DESCRIPTION
**Pull request by annibond**
Three new modules:
 - A concept assembly explaining the use of keycloak.
 - Procedure explaining installing dependencies,
 - Procedure explaining installing Keycloak itself.

Keycloak is a third party FOSS product that helps enable SSO in Ansible Automation Hub.

### CODE REVIEW

File: downstream/assemblies/central-auth/assembly-central-auth-automate-keycloak.adoc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[id="assembly-keycloak"]
+
+== Using Keycloak to automate single sign-on with {HubName}
+
+Keycloak is an open source tool developed by Red Hat that enables administrators to implement the use of SSO in software-as-a-service (SaaS) applications. It runs on top of Ansible using the collection and dependencies from Galaxy. With the base requirements running to install Keycloak, you can then install Keycloak and use it to automate the use of single sign-on with the use of Ansible playbooks.
```
Comment (ianf77) [inline downstream/assemblies/central-auth/assembly-central-auth-automate-keycloak.adoc:5]: Galaxy -> {Galaxy} which resolves as Ansible Galaxy

File: downstream/modules/platform/proc-central-auth-install-dependencies.adoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
+[id="proc-central-auth-dependencies"]
+
+= Install the collections and dependencies for Ansible Galaxy
```
Comment (ianf77) [inline downstream/modules/platform/proc-central-auth-install-dependencies.adoc:3]: {Galaxy}

File: downstream/modules/platform/proc-central-auth-install-dependencies.adoc [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+.Prerequisites
+
+. A supported version of either the https://access.redhat.com/product-life-cycles[Red Hat Enterprise Linux] or the https://docs.fedoraproject.org/en-US/releases/[Fedora] operating system
```
Comment (ianf77) [inline downstream/modules/platform/proc-central-auth-install-dependencies.adoc:7]: either link:https://access.redhat.com/product-life-cycles[{RHEL}] or the link:https://docs.fedoraproject.org/en-US/releases/[Fedora] operating system.

File: downstream/modules/platform/proc-central-auth-install-dependencies.adoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+. A supported version of https://access.redhat.com/support/policy/updates/ansible-automation-platform[Ansible]
```
Comment (ianf77) [inline downstream/modules/platform/proc-central-auth-install-dependencies.adoc:9]: of link:https://access.redhat.com/support/policy/updates/ansible-automation-platform[{PlatformNameShort}]

File: downstream/modules/platform/proc-central-auth-install-dependencies.adoc [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+.Procedure
+
+. Install the collection for Ansible Galaxy:
```
Comment (ianf77) [inline downstream/modules/platform/proc-central-auth-install-dependencies.adoc:13]: for {Galaxy}

File: downstream/modules/platform/proc-central-auth-install-dependencies.adoc [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
++
+[listing]
+$ ansible-galaxy collection install middleware_automation.keycloak
+
+[NOTE]
+====
+This collection has two dependencies that you might also want to install depending on your network configuration. 
+
+`middleware_automation.redhat_csp_download` enables Ansible to connect to the Customer Portal to download Red Hat’s single sign-on technology.
```
Comment (ianf77) [inline downstream/modules/platform/proc-central-auth-install-dependencies.adoc:22]: enables {PlatformNameShort} to ..

to download {RHSSO} technology

File: downstream/modules/platform/proc-central-auth-install-dependencies.adoc [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+`middleware_automation.wildfly` enables Keycloak to run on top of the Wildfly application server including Red Hat JBoss, the version of Wildfly officially supported by Red Hat.
```
Comment (ianf77) [inline downstream/modules/platform/proc-central-auth-install-dependencies.adoc:24]: I would check that Red Hat JBoss is  the official designation.

File: downstream/modules/platform/proc-central-auth-install-dependencies.adoc [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+====
+
+. Depending on your configuration, you might also want to add some Python dependencies containing libraries Ansible requires to use:
```
Comment (ianf77) [inline downstream/modules/platform/proc-central-auth-install-dependencies.adoc:27]: libraries used by {PlatformNameShort}:

File: downstream/modules/platform/proc-central-auth-install-keycloak.adoc [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+  vars:
+    keycloak_admin_password: "remembertochangeme"
+    keycloak_realm: TestRealm
+  collections:
+    - middleware_automation.keycloak
+  roles:
+    - keycloak
+
+
+The configuration adds Ansible’s collection for Keycloak to the list the playbook uses, and the associated middleware_automation.keycloak.keycloak.
```
Comment (ianf77) [inline downstream/modules/platform/proc-central-auth-install-keycloak.adoc:35]: adds {PlatformNameShort}'s collection... 

should  middleware_automation.keycloak.keycloak be in backticks?

File: downstream/modules/platform/proc-central-auth-install-keycloak.adoc [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+        	public_client: "{{ keycloak_client_public }}"
+        	web_origins: "{{ keycloak_client_web_origins }}"
+        	users: "{{ keycloak_client_users }}"
+        	client_id: TestClient
+
+[NOTE]
+====
+The playbook in Example 1 is the only playbook being used in this procedure. The following code blocks contain text from the same playbook. As a result, you are only modifying variables from one example. The examples following this note are truncated for findability and ease of understanding.
+
+In the Example 1, there are also no sources for authentication being used. This makes it easier to test your configuration before you deploy it. However, Red Hat recommends the use of LDAP as the authentication application.
```
Comment (ianf77) [inline downstream/modules/platform/proc-central-auth-install-keycloak.adoc:101]: In Exampe 1, there

**Review by ianf77 — COMMENTED** [review]
Just a few changes

File: downstream/modules/platform/proc-central-auth-install-dependencies.adoc (hunk 7 above)
Comment (annibond) [reply downstream/modules/platform/proc-central-auth-install-dependencies.adoc:24]: It is!

File: downstream/modules/platform/proc-central-auth-install-keycloak.adoc (hunk 9 above)
Comment (annibond) [reply downstream/modules/platform/proc-central-auth-install-keycloak.adoc:35]: yes it should thank you

File: downstream/modules/platform/proc-central-auth-install-keycloak.adoc (hunk 10 above)
Comment (annibond) [reply downstream/modules/platform/proc-central-auth-install-keycloak.adoc:101]: exampe ha! but thank you :)

**Pull request merged**